File: dotnet-restapi/InventoryManagementApi/DTOs/Process/ProcessDetail.cs
using InventoryManagementApi.DTOs.Step;
using InventoryManagementApi.DTOs.User;
using System;
using System.Collections.Generic;

namespace InventoryManagementApi.DTOs.Process
{
    public class ProcessDetail
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime CreationDate { get; set; }
        public string Type { get; set; }
        public string State { get; set; }
        public bool IsFavorite { get; set; }
        public int StepsCount { get; set; }
        public int FieldsCount { get; set; }
        public int UsersCount { get; set; }
        public DateTime? PublishingDate { get; set; }
        public DateTime? UnpublishingDate { get; set; }
        public string DataSource { get; set; }
        public ICollection<StepResponse> ProcessSteps { get; set; }
        public ICollection<UserDetail> Users { get; set; }
    }



}

File: angular-app/src/app/inventory-management/models/process/process-detail.ts

import { StepResponseDetail } from '../step/step-response-detail';
import { UserDetail } from '../user/user-detail';

export class ProcessDetail {
  id!: string;
  name!: string;
  description!: string;
  creationDate!: Date;
  type!: string;
  state!: string;
  isFavorite!: boolean;
  stepsCount!: number;
  fieldsCount!: number;
  usersCount!: number;
  publishingDate!: Date;
  unpublishingDate!: Date;
  dataSource!: string;
  processSteps: StepResponseDetail[] = [];
  users: UserDetail[] = [];
}
